tests/test_auxtel_detector_characterization_std_flat_dataset.py

```diff
@@ -67,14 +69,22 @@ async def cmd_take_images_callback(self, data):
         lexer.whitespace = ","
         parsed_data = dict(pair.split(":", 1) for pair in lexer)
 
-        if "bias" in parsed_data["imageType"].lower():
-            self.n_bias += 1
-        elif "dark" in parsed_data["imageType"].lower():
-            self.n_dark += 1
-        elif "flat" in parsed_data["imageType"].lower():
-            self.n_flat += 1
+        for i in range(data.numImages):
+            one_exp_time = data.expTime
+            if data.shutter:
+                one_exp_time += self.shutter_time
+            await asyncio.sleep(one_exp_time)
+
+            self.end_readout_tasks.append(asyncio.create_task(self.end_readout()))
+
+            if "bias" in parsed_data["imageType"].lower():
+                self.n_bias += 1
+            elif "dark" in parsed_data["imageType"].lower():
+                self.n_dark += 1
+            elif "flat" in parsed_data["imageType"].lower():
+                self.n_flat += 1
 
-        self.end_readout_tasks.append(asyncio.create_task(self.end_readout()))
+        await self.end_readout_tasks[-1]
```

Comment:
This suggests that you are sure that the tasks will complete in order. Perhaps consider a comment explaining that.